src/redux/missions/missionsSlice.js

```diff
@@ -0,0 +1,44 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const API_MISSION_URL = 'https://api.spacexdata.com/v3/missions';
+
+const initialState = {
+  missionItems: [],
+  loading: true,
+  error: undefined,
+};
+
+export const fetchMission = createAsyncThunk('missions/fetchMissions', async () => {
+  const response = await axios.get(API_MISSION_URL);
+  return response.data;
+});
+
+const missionsSlice = createSlice({
+  name: 'mission',
+  initialState,
+  reducers: {
+    addMission: (state, action) => {
+      console.log(action.payload);
+    },
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchMission.pending, (state) => {
+        state.loading = true;
+        state.error = undefined;
+      })
+      .addCase(fetchMission.fulfilled, (state, action) => {
+        const dataFromAPI = action.payload;
+        state.missionItems = dataFromAPI;
+      })
+      .addCase(fetchMission.rejected, (state, action) => {
+        state.loading = false;
+        state.error = action.error.message;
+      });
+  },
+});
+
+export const { addMission } = missionsSlice.actions;
+
+export default missionsSlice;
```

Comment:
As you have more than one export in the file, you can add `export` directly before `const` on line 17 and delete this line. This improves the code readability and matches well with JavaScript best practices.

The code will be like:
```
export const missionsSlice = createSlice({
  name: 'mission',
  initialState,
  reducers: {
    addMission: (state, action) => {
      console.log(action.payload);
    },
  },
```

[It's not mandatory but just a good practice]